ingest/event.go

```
@@ -0,0 +1,218 @@
+		return evt, ErrNotStellarAssetContract
+	}
+	fn := topics[0]
+
+	// Filter out events for function calls we don't care about
+	if fn.Type != xdr.ScValTypeScvSymbol {
+		return evt, ErrNotStellarAssetContract
+	}
+	if eventType, ok := STELLAR_ASSET_CONTRACT_TOPICS[fn.MustSym()]; !ok {
+		return evt, ErrNotStellarAssetContract
+	} else {
+		evt.Type = eventType
+	}
+
+	// This looks like a SAC event, but does it act like a SAC event?
+	//
+	// To check that, ensure that the contract ID of the event matches the
+	// contract ID that *would* represent the asset the event is claiming to
+	// be. The asset is in canonical SEP-11 form:
+	//  https://stellar.org/protocol/sep-11#alphanum4-alphanum12
+	//
+	// For all parsing errors, we just continue, since it's not a real
+	// error, just an event non-complaint with SAC events.
+	rawAsset := topics[len(topics)-1]
+	assetContainer, ok := rawAsset.GetObj()
+	if !ok || assetContainer == nil {
+		return evt, ErrNotStellarAssetContract
+	}
+
+	assetBytes, ok := assetContainer.GetBin()
+	if !ok || assetBytes == nil {
+		return evt, ErrNotStellarAssetContract
+	}
+
+	asset, err := txnbuild.ParseAssetString(string(assetBytes))
+	if err != nil {
+		return evt, ErrNotStellarAssetContract
+	}
+
+	xdrAsset, err := xdr.NewCreditAsset(asset.GetCode(), asset.GetIssuer())
+	if err != nil {
+		return evt, ErrNotStellarAssetContract
+	}
+	evt.Asset = xdrAsset
+
+	expectedId, err := xdrAsset.ContractID(networkPassphrase)
+	if err != nil {
+		return evt, errors.Wrap(ErrNotStellarAssetContract, err.Error())
+	}
+
+	// This is the DEFINITIVE integrity check for whether or not this is a
+	// SAC event. At this point, we can parse the event and treat it as
+	// truth, mapping it to effects where appropriate.
+	if expectedId != *event.ContractId { // nil check was earlier
+		return evt, ErrNotStellarAssetContract
+	}
+
+	switch evt.Type {
+	case EventTypeTransfer:
+		return evt, evt.parseTransferEvent(topics, value)
+	case EventTypeMint:
```

these seem to return an un-usable event, as only `Type` has been populated, maybe just throw unsupported also to avoid callers trying to do something, or do you think we can squeeze mint/clawback into the current `StellarAssetContractEvent`?

 I was wondering if mint is qualified for including in payments, I see payments includes `CreateAccount` operation, which is somewhat similar to mint?

---

As for payments, I think it should count, because in the Classic world, a "mint" of your custom asset only **_really_** happens after you make a `PaymentOp` to someone. Same thing with burns.